32 divergence from reference artwork

EricRLucas opened the pull request:
OK, I think I have finally figured out how it all works. After Nick sorted out the push to master, I ended up creating a new branch rather than continuing to work on the one Nick created, hence why my latest code push didn't end up on the other pull request thread.

EricRLucas commented:
PS: the colors in the figures are arbitrary for now, can easily be changed to be whatever population colours we use.

hardingnj commented:
Hi @EricRLucas . Sorry if I'm treading on your toes here, but I added a notebook that handles things in a slightly simpler way. There are also some tips regarding using the `agam-report-base module`.
As Alistair said on the call- we can use allele counts and some `scikit-allel` trickery. 
Happy to chat through it if helpful.

hardingnj commented:
I would expect the boxplot to be over windows rather than samples?
But boxplots are non parametric anyway?

EricRLucas commented:
My understanding of the boxplots is that it would replace the current figure in the sandbox (under the nucleotide variation section) where allelic diversity is plotted with each point representing a sample. We had discussed doing this for divergence from reference, and I had suggested doing a boxplot instead to summarise each population, instead of plotting every sample. A boxplot is indeed non-parametric, but you still need to calculate the median quartiles, which you can't do without individual-level data.

EricRLucas commented:
PS: I'm very happy to leave out the boxplots, but just pointing out that if we want to keep them, we'll need to calculate dxy by individual.

hardingnj commented:
Ah- ok with you now. I don't think looking at how different samples from the same populations look in terms of divergence from the reference is likely to yield anything interesting. They would all look very similar, and any differences would be driven by 2La/Rb karyotypes.

I envisaged something much more like:
> Could also plot divergence over the genome, per population, with maybe a couple of gambiae and a couple of coluzzii populations to compare. 
ie your second plot. 

I wouldn't mind making box plots, but I think they should be at the level of windows per population, rather than individuals. That way it highlights that some populations are missing regions highly diverged from PEST.

EricRLucas commented:
I'm not sure about plotting boxplots using the distribution across windows, as I'm not sure what it would tell us. The idea of boxplots taking individual-level data is just to give an idea of the scale of the difference between population relative to the variation that we see within a population. But I'm happy to forget the boxplots altogether.

For the line graphs, I think plotting all populations together gets confusing, because all of the lines overlap and hide each-other. I would say either have a line for each species, or plot each population on a separate plot (maybe a data-set wide line calculated using all populations could be included in each plot as a reference to help see which ones are lower and which ones are higher).

hardingnj commented:
> For the line graphs, I think plotting all populations together gets confusing, because all of the lines overlap and hide each-other. I would say either have a line for each species, or plot each population on a separate plot (maybe a data-set wide line calculated using all populations could be included in each plot as a reference to help see which ones are lower and which ones are higher).

Agreed, much better to split out or just have a representative few populations shown,

EricRLucas commented:
Hi @cclarkson @hardingnj , shall we have a Skype meeting this week to catch up on the phase 2 paper?

hardingnj commented:
1. You're exactly right- not needed. Must have been a hangover from when I used alelle counts in a previous version
2. It's just counting the proportions of each allele. As GATK calls everything as diploid, it won't fully account for this. We will therefore give males slight extra weight when computing allele frequencies.

hardingnj commented:
Seems reasonable to exclude males. We don't lose many samples, and they 
are in populations we can afford to lose them. Difference to results 
will be marginal, but for absolute correctness I think it makes sense. 
Chris?


On 31/01/2019 14:25, EricRLucas wrote:
>
> Cool, thanks @hardingnj <https://github.com/hardingnj> . Does it make 
> sense to just remove the males then? Or are we happy that the results 
> won't be far wrong? Also tagging @cclarkson 
> <https://github.com/cclarkson> for your opinion.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/malariagen/ag1000g-phase2-data-paper/pull/67#issuecomment-459362760>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AEO2zOEnxXT-PLoZ-HGMjkpJHEOWFHnPks5vIvzkgaJpZM4ZAMAM>.
>
-- 
Nicholas Harding
+44 (0) 1865 617786 / @nick_harding
NOTE: change of email: now nicholas.harding@bdi.ox.ac.uk
Global Health and Malaria Programme
Kwiatkowski Group
Big Data Institute
Li Ka Shing Centre for Health Information and Discovery
University of Oxford

-

Research Member of Common Room
Kellogg College, Oxford

cclarkson commented:
I think either way is fine, but removing males seems more defensible.

alimanfoo commented:
+1 on removing males.

On Fri, 1 Feb 2019 at 11:33, Chris Clarkson <notifications@github.com>
wrote:

> I think either way is fine, but removing males seems more defensible.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/malariagen/ag1000g-phase2-data-paper/pull/67#issuecomment-459678995>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAq8QjFmWMqKlBWE6sb3y6Ll3cknrsxGks5vJBf_gaJpZM4ZAMAM>
> .
>


-- 

Alistair Miles
Head of Epidemiological Informatics
Centre for Genomics and Global Health
Big Data Institute
Li Ka Shing Centre for Health Information and Discovery
University of Oxford
Old Road Campus
Headington
Oxford
OX3 7LF
United Kingdom
Phone: +44 (0)1865 743596 or +44 (0)7866 541624
Email: alimanfoo@googlemail.com
Web: http://a <http://purl.org/net/aliman>limanfoo.github.io/
Twitter: @alimanfoo <https://twitter.com/alimanfoo>

Please feel free to resend your email and/or contact me by other means if
you need an urgent reply.

EricRLucas commented:
@hardingnj @cclarkson  I have recreated the plots after removing males (I removed males for all chromosomes, otherwise we have the slightly odd situation where within a single plot, different individuals were used to generate the data from different parts of the genome). I will start writing the text, but I'll do this in a different branch as it's got a different issue number. Once this PR is merged, I'll be able to pull the plots into the tex document.

EricRLucas commented:
@alimanfoo Here are the Dxy plots for various populations / species. While trying to interpret them, we figured that any pattern could have two causes: increased diversity, or just difference from the reference genome. The only way to distinguish between them is through a priori ideas about what should be causing the signal (eg: drop in diversity around centromeres and 2La). What were we hoping to use these plots to show?

EricRLucas commented:
@cclarkson @hardingnj I've created the plot of Dxy between each of the unkown species pops and each of coluzzii and gambiae? They're in the file phase2_refdiff_combied.jpg. Is this what we expected them to look like though? I attempted to use scikit-allel to make the calculations, so feel free to review my code and check I did it right.

EricRLucas commented:
I've removed the bottom three plots from the main figure. I've also removed the legend title, as "Species" didn't accurately describe the fact that subsequent plots actually represented different populations, and I think the labels are self-evident enough to not require a title.
